src/lib.rs

```diff
@@ -1283,6 +1302,50 @@ impl Plugin for EguiPlugin {
                     Render,
                     render::systems::queue_pipelines_system.in_set(RenderSet::Queue),
                 );
+
+            // Configure a fixed rendering order between Bevy UI and egui.
+            // Otherwise, this order is effectively decided at random on every game startup.
+            #[cfg(feature = "bevy_ui")]
+            {
+                use bevy_render::render_graph::RenderLabel;
+                let mut graph = render_app
+                    .world_mut()
+                    .resource_mut::<bevy_render::render_graph::RenderGraph>();
+                let (below, above) = match self.ui_render_order {
+                    UiRenderOrder::EguiAboveBevyUi => (
+                        bevy_ui::graph::NodeUi::UiPass.intern(),
+                        render::graph::NodeEgui::EguiPass.intern(),
+                    ),
+                    UiRenderOrder::BevyUiAboveEgui => (
+                        render::graph::NodeEgui::EguiPass.intern(),
+                        bevy_ui::graph::NodeUi::UiPass.intern(),
+                    ),
+                };
+                if let Some(graph_2d) =
+                    graph.get_sub_graph_mut(bevy_core_pipeline::core_2d::graph::Core2d)
+                {
+                    // Only apply if the bevy_ui plugin is actually enabled.
+                    // In theory we could use RenderGraph::try_add_node_edge instead and ignore the result,
+                    // but that still seems to end up writing the corrupt edge into the graph,
+                    // causing the game to panic down the line.
+                    if graph_2d
+                        .get_node_state(bevy_ui::graph::NodeUi::UiPass)
+                        .is_ok()
```

Comment:
I think we should add a warning here if we fail to add the edge, to explain to a user why their config hasn't been applied

Reply:
Yeah I was going back and forth on this.. it's a weird and kind of surprising error condition.. but on the other hand, that would end up logging warnings for every project that disables bevy_ui (which seems like a natural thing to do if you use egui for all your UI needs).